tests/suites/test_suite_pk.function

```diff
@@ -233,6 +237,94 @@ static int pk_genkey(mbedtls_pk_context *pk, int curve_or_keybits)
     return ret;
 }
 
+/** Create a PSA key of the desired type and properties.
+ *
+ * This is similar to pk_genkey() above in the sense that it does not really
+ * generates a key every time, but it takes the key from a file instead in
+ * order to speedup testing.
+ *
+ * \param type                  PSA key type. Only RSA and EC keys are supported.
+ * \param bits                  PSA key bit size.
+ * \param usage                 PSA key usage flags.
+ * \param alg                   PSA key primary algorithm.
+ * \param enrollment_alg        PSA key enrollment algorithm.
+ * \param persistent_key_id     PSA key ID for persistent keys. Set to PSA_KEY_ID_NULL
+ *                              for volatile keys.
+ * \param[out] key              Identifier of the "generated" (actually imported) PSA key.
+ */
+psa_status_t pk_psa_genkey(psa_key_type_t type, size_t bits,
+                           psa_key_usage_t usage, psa_algorithm_t alg,
+                           psa_algorithm_t enrollment_alg,
+                           mbedtls_svc_key_id_t persistent_key_id,
+                           mbedtls_svc_key_id_t *key)
+{
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    mbedtls_pk_context pk;
+    char file_name[128] = { 0 };
+    unsigned char key_data[MBEDTLS_PK_WRITE_PUBKEY_MAX_SIZE] = { 0 };
+    size_t key_data_len;
+    unsigned char *key_data_start;
+    int ret;
+
+    mbedtls_pk_init(&pk);
+
+    /* Get the name of the key file to load. */
+    if (PSA_KEY_TYPE_IS_RSA(type)) {
+        sprintf(file_name, "data_files/rsa_%lu.der", bits);
+    } else if (PSA_KEY_TYPE_IS_ECC(type)) {
+        psa_ecc_family_t ec_family = PSA_KEY_TYPE_ECC_GET_FAMILY(type);
+        mbedtls_ecp_group_id grp_id = mbedtls_ecc_group_from_psa(ec_family, bits);
+        sprintf(file_name, "data_files/ec_%s.der", curve_names_lut[grp_id]);
+    } else {
+        TEST_FAIL("Only EC or RSA key type is supported.");
+    }
+    /* Parse the key file and write the key material to the key_data buffer. */
+    TEST_EQUAL(mbedtls_pk_parse_keyfile(&pk, file_name, NULL, mbedtls_test_rnd_std_rand, NULL), 0);
+    if (mbedtls_pk_get_type(&pk) == MBEDTLS_PK_RSA) {
```

Comment:
I'm not sure I get this part. Now that we have a PK object can't we just use `pk_import_into_psa()`?

Also, same reservation as before about using a file: I'd rather have a plain C array. Also, while at it perhaps we can have the generation script give use the key in PSA format so that we can directly call `psa_import_key()` without even depending on `PK_PARSE_C`.